process_trees.py

```diff
@@ -0,0 +1,94 @@
+from collections import deque
+import pickle
+#
+import torch
+import bcolz
+from torchtext.data.utils import ngrams_iterator
+from torchtext.data.utils import get_tokenizer
+#
+from reddit_data import RedditTrees, JsonDataSource
+
+UNKNOWN_WORD = "[unk]"
+
+# Setup glove embedding
+glove_path = "machine_learning/glove"
+
+word_vectors = bcolz.open(f'{glove_path}/6B.50.dat')[:]
+words = pickle.load(open(f'{glove_path}/6B.50_words.pkl', 'rb'))
+word2idx = pickle.load(open(f'{glove_path}/6B.50_idx.pkl', 'rb'))
+
+glove = {w: word_vectors[word2idx[w]] for w in words}
+tokenizer = get_tokenizer("basic_english")
+
+# Depth first search as a generator
+def dfs(tree):
+    queue = deque()
+    queue.append(tree)
+    while(queue):
+        node = queue.popleft()
+        yield node
+        for child in node.children:
+            queue.append(child)
+
+def create_adj_matrix(edges, id_to_index):
+    def add_edge(node):
+        index = id_to_index[node.comment.id]
+        for child in node.children:
+            child_index = id_to_index[child.comment.id]
+            edges.append([index, child_index])
+    return add_edge
+
+def get_tree_indices(tree):
+    return {node.comment.id: index for index, node in enumerate(dfs(tree))}
+
+# helper method to create an ajacency matrix given a mapping from each node to an index
+# The id_to_index mapping is passed to simplify the logic
+def get_adj_matrix(tree, id_to_index):
+    node_edges = []
+    for node in dfs(tree):
```

Comment:
Given that this loop goes thru the tree in the same order as indexes are assigned in get_tree_indices(), I'm wondering if looping through the tree twice and saving the indices from the first run is actually necessary.

Reply:
Unless those indices are needed for later?

Reply:
Perhaps a solution would be to put an `index` field on the RedditNode and assign that information at tree building time.